### PR TITLE
Modal component, payment cancelling, processing errors in popovers and other improvements

### PR DESCRIPTION
**Pull request by MadOPcode**
Implements #54 #56 #57 #61.

### CODE REVIEW

File: src/components/Modal/index.jsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import styles from "./styles.module.scss";
+import "react-responsive-modal/styles.css";
+
+const classNames = {
+  modal: styles.modal,
+  modalContainer: styles.modalContainer,
+};
+const closeIcon = <IconCross />;
+
+const Modal = ({
```
Comment (maxceem) [inline src/components/Modal/index.jsx:16]: Can we also add comment docs for this component?

File: src/routes/WorkPeriods/components/PaymentCancel/index.jsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+import Modal from "components/Modal";
+import Spinner from "components/Spinner";
+import { makeToast } from "components/ToastrMessage";
+import { PAYMENT_STATUS } from "constants/workPeriods";
+import { setWorkPeriodPaymentData } from "store/actions/workPeriods";
+import { cancelWorkPeriodPayment } from "services/workPeriods";
+import styles from "./styles.module.scss";
+import { loadWorkPeriodAfterPaymentCancel } from "store/thunks/workPeriods";
+
+const PaymentCancel = ({ className, item, timeout = 3000 }) => {
```
Comment (maxceem) [inline src/routes/WorkPeriods/components/PaymentCancel/index.jsx:15]: Missing comment docs.

**Review by maxceem — CHANGES_REQUESTED** [review]
1. On this page all the rows become higher and there is a horizontal scrollbar even though on other pages there is no scrollbar and rows has smaller height https://platform.topcoder-dev.com/taas-admin/work-periods?startDate=2021-06-27&by=user_handle&order=asc&perPage=10&page=1

    image

    Before applying this PR it looked good:

    image

2. Sorry, I've missed this case. Let's hide the `Cancel` button when payment is canceled:

     image

File: src/components/Modal/index.jsx (hunk 1 above)
Comment (MadOPcode) [reply src/components/Modal/index.jsx:16]: > 1. On this page all the rows become higher...

On my local machine I don't see a scrollbar and the rows have the correct height.
https://monosnap.com/file/B1fcFmmp2aqc9HB4OIdnxB4xVnqEsf
So my first guess is that something from navbar/frame influences the taas-admin styles on platform.topcoder-dev.com, considering that I haven't updated navbar/frame for quite a while.

**Comment by maxceem** [comment]
@MadOPcode I pushed small fixes for https://github.com/topcoder-platform/micro-frontends-taas-admin-app/issues/48 and it causes merge conflicts now. Could you, please, pull the latest changes from `dev` branch to resolve merge conflicts for this PR.

File: src/components/Modal/index.jsx (hunk 1 above)
Comment (maxceem) [reply src/components/Modal/index.jsx:16]: I'm testing this PR with the same version of the Navbar. So the only I change is which verions of Admin App I run.
If I run Admin App before this PR all good. If I apply this PR and run Admin App with other things all the same, the height sometimes goes up. It seems that It's caused by the `processingError` span, see screenshot below:

image

Comment (maxceem) [reply src/components/Modal/index.jsx:31]: @MadOPcode I did more testing, the height got increased only when the screen is relatively small, see demo video https://monosnap.com/file/QUtc4EZJgjYH9F3Sx4jM6HXnbE1EvF

Comment (MadOPcode) [reply src/components/Modal/index.jsx:31]: I think I got it.

File: src/routes/WorkPeriods/components/PaymentCancel/index.jsx (hunk 2 above)
Comment (MadOPcode) [reply src/routes/WorkPeriods/components/PaymentCancel/index.jsx:26]: All fixed.

**Review by maxceem — APPROVED** [review]
@MadOPcode all works perfectly now.

**Pull request merged**